### DISCUSSION TITLE
difference between integrated data from SCTransform with and without return.only.var.genes=TRUE

### DISCUSSION THREAD

**Discussion by nyanmo** [satijalab/seurat#4973] [Q&A]
Hi, thank you for the great tool.
I have a question about Integrated data.
I normalized datasets with SCTransform, where return.only.var.genes were set TRUE or FALSE.
obj1T <- SCTransform( obj1, return.only.var.genes = T );
obj2T <- SCTransform( obj2, return.only.var.genes = T );
obj1F <- SCTransform( obj1, return.only.var.genes = F );
obj2F <- SCTransform( obj2, return.only.var.genes = F );

After I integrated obj1T and obj2T, and obj1F and obj2F, I saw that integrated@scale.data were different.
I would like to ask you which normalization is recommended.

**Comment by saketkc** [comment]
As explained in the [doc](https://satijalab.org/seurat/reference/sctransform), `return.only.var.genes=T` which is set to TRUE by default will only create the scale.data slot (pearson residuals) for genes that are highly variable.

**Reply by nyanmo** [reply]
> As explained in the [doc](https://satijalab.org/seurat/reference/sctransform), `return.only.var.genes=T` which is set to TRUE by default will only create the scale.data slot (pearson residuals) for genes that are highly variable.
Thank you for your comments. I understand the default value of return.only.var.genes.

After integrating objects, I saw that the scale.data of cobjT and cobjF were different:
listT <- list( obj1T, obj2T );
listF <- list( obj1F, obj2F );

feT <- SelectIntegrationFeatures( object.list = listT );
feF <- SelectIntegrationFeatures( object.list = listF );

listT <- PrepSCTIntegration( object.list = listT, anchor.features = feT );
listF <- PrepSCTIntegration( object.list = listF, anchor.features = feF );

anchT <- FindIntegrationAnchors( object.list = listT, anchor.features = feT );
anchF <- FindIntegrationAnchors( object.list = listF, anchor.features = feF );

cobjT <- IntegrateData( anchorset = anchT );
cobjF <- IntegrateData( anchorset = anchF );

My question is whether setting `return.only.var.genes=F` is recommended when executing SCTransform before SelectIntegrationFeatures or not.
Are scale data values that were not calculated at the SCTransform calculated when integrating objects?

**Reply by saketkc** [reply]
Only variable features are used in [SelectIntegrationFeatures](https://satijalab.org/seurat/reference/selectintegrationfeatures). `return.only.var.genes=F` will create the `scale.data` slot for all features but only variable features are used downstream.
